cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java

```diff
@@ -42,25 +41,21 @@
 public final class HBaseStreamConsumerStateStoreFactory implements StreamConsumerStateStoreFactory {
 
   private final Configuration hConf;
-  private final String storeTableName;
   private final HBaseTableUtil tableUtil;
-  private boolean tableCreated;
 
   @Inject
-  HBaseStreamConsumerStateStoreFactory(Configuration hConf, CConfiguration conf, HBaseTableUtil tableUtil) {
+  HBaseStreamConsumerStateStoreFactory(Configuration hConf, HBaseTableUtil tableUtil) {
     this.hConf = hConf;
-    this.storeTableName =
-      HBaseTableUtil.getHBaseTableName(new DefaultDatasetNamespace(conf, Namespace.SYSTEM)
-                                         .namespace((QueueConstants.STREAM_TABLE_PREFIX) + ".state.store"));
     this.tableUtil = tableUtil;
   }
 
   @Override
   public synchronized StreamConsumerStateStore create(StreamConfig streamConfig) throws IOException {
-    byte[] tableName = Bytes.toBytes(storeTableName);
+    Id.Namespace namespace = streamConfig.getStreamId().getNamespace();
+    byte[] tableName = Bytes.toBytes(constructTableName(namespace));
 
-    if (!tableCreated) {
-      HBaseAdmin admin = new HBaseAdmin(hConf);
+    HBaseAdmin admin = new HBaseAdmin(hConf);
+    if (!admin.tableExists(tableName)) {
```

Comment:
As discussed offline, will use hbase namespaces for all tables in a separate PR (rather than use hbase namespaces for simply this table).